examples/energy-management-app/energy-management-common/src/DeviceEnergyManagementDelegateImpl.cpp

```diff
@@ -323,49 +857,140 @@ CHIP_ERROR DeviceEnergyManagementDelegate::SetESAState(ESAStateEnum newValue)
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR DeviceEnergyManagementDelegate::SetAbsMinPower(int64_t newValue)
+CHIP_ERROR DeviceEnergyManagementDelegate::SetAbsMinPower(int64_t newValueMw)
 {
-    int64_t oldValue = mAbsMinPower;
+    int64_t oldValueMw = mAbsMinPowerMw;
 
-    mAbsMinPower = newValue;
-    if (oldValue != newValue)
+    mAbsMinPowerMw = newValueMw;
+    if (oldValueMw != newValueMw)
     {
-        ChipLogDetail(AppServer, "mAbsMinPower updated to %d", static_cast<int>(mAbsMinPower));
+        ChipLogDetail(AppServer, "mAbsMinPower updated to %d", static_cast<int>(mAbsMinPowerMw));
         MatterReportingAttributeChangeCallback(mEndpointId, DeviceEnergyManagement::Id, AbsMinPower::Id);
     }
 
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR DeviceEnergyManagementDelegate::SetAbsMaxPower(int64_t newValue)
+CHIP_ERROR DeviceEnergyManagementDelegate::SetAbsMaxPower(int64_t newValueMw)
 {
-    int64_t oldValue = mAbsMaxPower;
+    int64_t oldValueMw = mAbsMaxPowerMw;
 
-    mAbsMaxPower = newValue;
-    if (oldValue != newValue)
+    mAbsMaxPowerMw = newValueMw;
+    if (oldValueMw != newValueMw)
     {
-        ChipLogDetail(AppServer, "mAbsMaxPower updated to %d", static_cast<int>(mAbsMaxPower));
+        ChipLogDetail(AppServer, "mAbsMaxPower updated to %d", static_cast<int>(mAbsMaxPowerMw));
```

Comment:
This is probably worth documentation about why the cast is safe (raising questions about why the underlying value is 64-bit, but OK).  Same in other places where this cast is happening.

Reply:
The cast has been replaced with ChipLogFormatX64/ChipLogValueX64

Reply:
@PeterC1965 I don't see how that makes sense: ChipLogFormatX64 assumes an unsigned 64-bit value, but these are signed.

Reply:
Closing per Boris' approval on call.